lib/exirc/utils.ex

```diff
@@ -25,7 +25,7 @@ defmodule ExIrc.Utils do
     end
   end
 
-  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?:(?<user>[^@ ]+)@)?(?:(?<host>[\w.:-]+)))?$/
+  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?:(?<user>[^@ ]+)@)?(?:(?<host>\S+[\w.:-]+)))?$/
```

Comment:
From our conversations on IRC, the one without nick or user improvements works, so try adding in the improved nick matching, then if that works try adding user matching as well. If not, disable it and try user matching.

That way, I can figure out where the issue(s) is/are.